wom/src/test/scala/wom/types/WomMapTypeSpec.scala

```diff
@@ -57,14 +57,47 @@ class WomMapTypeSpec extends FlatSpec with Matchers  {
       case Failure(_) => fail("Failed to coerce a map to an object")
     }
   }
-  it should "detect invalid map construction if there are mixed types" in {
+
+  it should "detect invalid map construction if there are mixed keys" in {
     try {
-      WomMap(WomMapType(WomStringType, WomStringType), Map(WomInteger(0) -> WomString("foo"), WomString("x") -> WomInteger(2)))
+      WomMap(WomMapType(WomStringType, WomStringType), Map(WomInteger(0) -> WomString("foo"), WomString("x") -> WomString("y")))
       fail("Map initialization should have failed")
     } catch {
-      case _: UnsupportedOperationException => // expected
+      case e: UnsupportedOperationException =>
+        e.getMessage shouldEqual "Cannot construct WomMapType(WomStringType,WomStringType) with mixed types in map keys: [WomInteger(0), WomString(x)]"
```

Comment:
Should map these to the value Strings to avoid leaking these `WomX()` wrappers

Reply:
I think the string coercion might strip out crucial information, such as eliminating the visible difference between `WomSingleFile("/foo/bar.wdl")` and `WomString("/foo/bar.wdl")`